lib/configuration.ml

```diff
@@ -77,23 +97,18 @@ module ParseConfig = struct
     let insert_append k v =
       Dict.update k (fun curr -> Some (v :: Option.value curr ~default:[]))
     in
-    let add_line_num i =
-      Error.map_msg (fun msg -> Printf.sprintf "Entry starting at Line %d, %s" i msg)
-    in
     let update_accum line_num next accum =
-      Result.witherr (add_line_num line_num) (entry_of_assoc next) >>= (fun entry ->
+      Result.witherr (fun k -> `ConfigData (line_num, k)) (entry_of_assoc next) >>= (fun entry ->
         Ok (insert_append entry.source_type (transform_data_of_entry entry) accum))
     in
     let collect_varieties line_num next raccum = raccum >>= update_accum line_num next in
-    let error_handler line_num line rerror     = rerror >>= (fun _ ->
-      Error (`ReferParse
-        (Printf.sprintf "Line %d, Refer Parse Error: Cannot parse '%s'" line_num line)))
+    let error_handler     line_num line rerror = rerror >>= (fun _ ->
+      Error (`ReferParse (line_num, line)))
     in
     Refer.fold
       (Refer.witherr error_handler collect_varieties)
       (Ok Dict.empty)
       (Refer.of_string config_str)
 
-  let parse_config_file path_to_config =
-    path_to_config |> Prelude.readfile |> parse_config_str
+  let parse_config_file = Prelude.readfile >> parse_config_str
```

Comment:
No compiler error here, so going pointfree is great.  However, I want to note that OCaml's type inference will sometimes choke on pointfree definitions because it gets stuck trying to promote those weak polymorphic variables to real polymorphic variables.  (Weak polymorphic variables, in the sense of that Garrigue paper I sent you.)  Kind of annoying.  But fixable in every case just by eta expanding the definition.  Just FYI, in case you come across that.